### PR TITLE
JBRes-1959 Refactor Project Transformation/Compilation

### PR DESCRIPTION
**Pull request by khbminus**
## Title (hint)

The title's format is following: `JBRes-#### <Some description>`

The title should contain a related ticket number from YT, and a short description of your changes or a new functionality.

For example:
> JBRes-1877 Kotlinc Exceptions Transformers

## Description

- Added a `ProjectItemLens` for focusing on the chosen project items in each context

## How to test

I see it as a simple refactor in the current code base, so no new tests are needed here

### Automated tests

Verified with `GradleCompilationTest.kt` and `MinimizationServiceTest` (as integration). 

### Manual tests

Not Needed

## Self-check list

- [x] PR **title** and **description** are clear and aligned with a format.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas. 
- [x] The functionality I've repaired, changed or added is covered with **automated tests**.
- [x] **Manual tests** have been provided _optionally_.
- [x] The **documentation** for the functionality I've been working on is up-to-date/provided.
- [x] The **link** to this PR is commented on in the corresponding **YT ticket**.

Hint: [x] is a marked item

### CODE REVIEW

File: project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/ProjectItemLens.kt
Comment (SBOne-Kenobi) [inline project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/ProjectItemLens.kt:7]: I don't know (I didn't get it), but the class and function names are strange. It's more about project transformation/DDItems application than focusing on something. What do you think?

File: project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginState.kt
Comment (SBOne-Kenobi) [inline project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginState.kt]: I think it's better to create it on demand inside algorithms without providing any settings because it's closely connected with the algorithm

File: project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/ProjectItemLens.kt
Comment (khbminus) [reply project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/ProjectItemLens.kt:7]: IMO, the main usage is to reduce the project items to a list of selected ones => using an interface focus on the specific part of the project. So it looks like a lens (not the Functional Programming one, yea)

**Comment by SBOne-Kenobi** [comment]
Nah, I [said](https://github.com/plan-research/project-minimization/pull/17#discussion_r1812127436) to delete the new setting `propertyCheckerProjectModificationLens`. Settings about transformations were useful (at least for tests)

**Comment by khbminus** [comment]
Fuck, I'm stupid. Sorry :)

**Review by SBOne-Kenobi — APPROVED** [review]
LGTM, please wait for #12 before merging :) I hope it will be merged soon

**Comment by SBOne-Kenobi** [comment]
Ping @khbminus

**Pull request merged**
